inst/EpiModel 2.0 Introduction.rmd

```diff
@@ -0,0 +1,254 @@
+
+### Set Functions
+
+The ```set``` group of functions sets the value of an item on a sublist of the ```dat``` object; these functions are often used at the end of a modules sequence to update attributes or for tracking of epidemic statistics. As an example: 
+
+```{r echo = TRUE, eval = FALSE}
+## Code snippet from one-Group infection module:
+dat <- set_epi(dat, "si.flow", at, nInf)
+```
+
+Here, we want to record the number of new infected - ```nInf``` -  at the current time step ```at```. We do so using the ```set_epi``` helper function and record the flow of succeptible to infected on to the master ```dat``` object. All set functions follow this naming scheme, with "set_" followed by the sublist of interest. Note: the set functions returns an updated ```dat``` object.
+
+### Get Functions
+
+The ```get``` group of functions retrieve the value of an item on a sublist of the ```dat``` objects, and are most often used when retrieving input parameters of ```init```, ```param``` and ```control``` objects or for querying current attribute or epidemic tracker lists. As an example:
+
+```{r echo = TRUE, eval = FALSE}
+## Code snippet from One-Group arrivals module
+a.rate <- get_param(dat, "a.rate")
+nOld <- get_epi(dat, "num", index)
+```
+
+Here we are pulling the arrival rate, ```a.rate``` from the parameter list and querying the epidemic tracker list for the population size at the previous step. Note: the get functions will return a value or a vector of values as oppossed to the ```dat``` object returned by the set functions.
+
+Details of all available helper functions can be found using
+
+```{r echo = TRUE, eval = FALSE}
+?EpiModel::dat_get_set
+```
+
+# Pre-EpiModel 2.0 Code Migration
+
+The following section will help EpiModel users update code written in the EpiModel 1.0 language to EpiModel 2.0.
+
+## Network_initialize replaces network.initialize
+
+New ```tergmLite``` version of network initilization function:
+
+```{r, echo = TRUE, eval = FALSE}
+nw <- network.initialize(num = 100)
+```
+
+is updated to
+
+```{r, echo = TRUE, eval = FALSE}
+nw <- network_initialize(num = 100)
+```
+
+Similarly, ```set.vertex.attribute``` and ```get.vertex.attribute``` have beenu updated as well:
+```{r echo = TRUE, eval = FALSE}
+nw <- set.vertex.attribute(nw, "group", rep(1:2, 50))
+group <- get.vertex.attribute(nw, "group")
+```
+
+is updated to
+
+```{r echo = TRUE, eval = FALSE}
+nw <- set_vertex_attribute(nw, "group", rep(1:2, 50))
```

Comment:
make sure you note above that group is now a special attribute, along with status. When networks contain these attributes EpiModel will behave differently than other attribute names. Explain that in detail above.

Reply:
Added warning regarding "group" as a vertex attribute.